app/src/main/java/com/example/drcreditdev/creditCardUI/refreshPageUI/RefreshPage.kt

```diff
@@ -24,55 +20,87 @@ class RefreshPage : AppCompatActivity() {
 
     var diff : String? = null
     private lateinit var sharedPreferences: SharedPreferences
-    private lateinit var apiClient: ApiClientRefresh
+    lateinit var dataBinding : ActivityRefreshPageBinding
+    var viewModel: RefreshPageViewModel = RefreshPageViewModel()
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_refresh_page)
-        sharedPreferences = getSharedPreferences("drFile", MODE_PRIVATE)
-        var authToken:String? = sharedPreferences.getString("authToken","user_token")
-        val header = HashMap<String, String>()
-        header["authToken"] = authToken!!
-        header["Content-Type"] = "application/json"
-        header["Cookie"] = "JSESSIONID=F7AC6F52A47B66466769E1EFE88BBA9B"
-
-        val retrofitBuilder = Retrofit.Builder()
-            .addConverterFactory(GsonConverterFactory.create())
-            .baseUrl("https://stage.terrafin.tech:8090/underwriting/").build()
-            .create(RetrofitApi::class.java)
-
-        val retrofitData = retrofitBuilder.getRefreshScore(header)
+        dataBinding = ActivityRefreshPageBinding.inflate(layoutInflater)
+
+        setContentView(dataBinding.root)
+        dataBinding.viewModel = viewModel
+        viewModel = ViewModelProvider(this).get(RefreshPageViewModel::class.java)
+
+        changePage()
+
+    }
+    private fun changePage() {
+
         val c = Calendar.getInstance()
-        val year : String = c.get(Calendar.YEAR).toString()
+        val year: String = c.get(Calendar.YEAR).toString()
         val month = c.get(Calendar.MONTH)
         val day = c.get(Calendar.DAY_OF_MONTH)
 
-          var len = year.length
-          yearS = (year.get(len-1)+""+year.get(len-2))
-        monthS = when(month){
-            0 ->"Jan"
-            1->"Feb"
-            2->"Mar"
-            3->"Apr"
-            4->"May"
-            5->"June"
-            6->"Jul"
-            7->"Aug"
-            8->"Sept"
-            9->"Oct"
-            10->"Nov"
-            11->"Dec"
+        var len = year.length
+        yearS = (year.get(len - 1) + "" + year.get(len - 2))
+        monthS = when (month) {
+            0 -> "Jan"
+            1 -> "Feb"
+            2 -> "Mar"
+            3 -> "Apr"
```

Comment:
move to utils

Reply:
tried it! But couldn't handle the errors! Will do in next commit